lib/OpenQA/Shared/Plugin/Gru.pm

```diff
@@ -45,13 +45,14 @@ sub register ($self, $app, $config) {
     my $schema = $app->schema;
 
     my $conn = Mojo::Pg->new;
-    if (ref $schema->storage->connect_info->[0] eq 'HASH') {
+    my $connect_info = $schema->storage->connect_info->[0];
+    if (ref $connect_info eq 'HASH') {
         $self->dsn($schema->dsn);
```

Comment:
This change removes `$schema->dsn`, so how is this still supposed to work?

Reply:
Maybe I lost that change by now by rebasing over the course of two years :) But maybe you have a good idea what we can do to introduce proper test coverage here now?

Reply:
We should probably just invoke the function with `$schema->storage->connect_info` being a hash (as I think this is the missing branch). We might need to mock `Mojo::Pg` and possibly other packages.

Reply:
Wanting to try out to call openqa-gru manually I got inspired to draft
* #5796

Reply:
I really meant to invoke this single function.

Reply:
I suspect that we can simply delete the if-branch